// ngx-highlightjs/esm5/plus/gist.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { InjectionToken } from '@angular/core';
/**
 * @record
 */
export function GistOptions() { }
if (false) {
    /** @type {?} */
    GistOptions.prototype.clientId;
    /** @type {?} */
    GistOptions.prototype.clientSecret;
}
/** @type {?} */
export var GIST_OPTIONS = new InjectionToken('GIST_OPTIONS');
/**
 * @record
 */
function Owner() { }
if (false) {
    /** @type {?} */
    Owner.prototype.login;
    /** @type {?} */
    Owner.prototype.id;
    /** @type {?} */
    Owner.prototype.node_id;
    /** @type {?} */
    Owner.prototype.avatar_url;
    /** @type {?} */
    Owner.prototype.gravatar_id;
    /** @type {?} */
    Owner.prototype.url;
    /** @type {?} */
    Owner.prototype.html_url;
    /** @type {?} */
    Owner.prototype.followers_url;
    /** @type {?} */
    Owner.prototype.following_url;
    /** @type {?} */
    Owner.prototype.gists_url;
    /** @type {?} */
    Owner.prototype.starred_url;
    /** @type {?} */
    Owner.prototype.subscriptions_url;
    /** @type {?} */
    Owner.prototype.organizations_url;
    /** @type {?} */
    Owner.prototype.repos_url;
    /** @type {?} */
    Owner.prototype.events_url;
    /** @type {?} */
    Owner.prototype.received_events_url;
    /** @type {?} */
    Owner.prototype.type;
    /** @type {?} */
    Owner.prototype.site_admin;
}
/**
 * @record
 */
function User() { }
if (false) {
    /** @type {?} */
    User.prototype.login;
    /** @type {?} */
    User.prototype.id;
    /** @type {?} */
    User.prototype.node_id;
    /** @type {?} */
    User.prototype.avatar_url;
    /** @type {?} */
    User.prototype.gravatar_id;
    /** @type {?} */
    User.prototype.url;
    /** @type {?} */
    User.prototype.html_url;
    /** @type {?} */
    User.prototype.followers_url;
    /** @type {?} */
    User.prototype.following_url;
    /** @type {?} */
    User.prototype.gists_url;
    /** @type {?} */
    User.prototype.starred_url;
    /** @type {?} */
    User.prototype.subscriptions_url;
    /** @type {?} */
    User.prototype.organizations_url;
    /** @type {?} */
    User.prototype.repos_url;
    /** @type {?} */
    User.prototype.events_url;
    /** @type {?} */
    User.prototype.received_events_url;
    /** @type {?} */
    User.prototype.type;
    /** @type {?} */
    User.prototype.site_admin;
}
/**
 * @record
 */
function ChangeStatus() { }
if (false) {
    /** @type {?} */
    ChangeStatus.prototype.total;
    /** @type {?} */
    ChangeStatus.prototype.additions;
    /** @type {?} */
    ChangeStatus.prototype.deletions;
}
/**
 * @record
 */
function History() { }
if (false) {
    /** @type {?} */
    History.prototype.user;
    /** @type {?} */
    History.prototype.version;
    /** @type {?} */
    History.prototype.committed_at;
    /** @type {?} */
    History.prototype.change_status;
    /** @type {?} */
    History.prototype.url;
}
/**
 * @record
 */
function Files() { }
/**
 * @record
 */
export function Gist() { }
if (false) {
    /** @type {?} */
    Gist.prototype.url;
    /** @type {?} */
    Gist.prototype.forks_url;
    /** @type {?} */
    Gist.prototype.commits_url;
    /** @type {?} */
    Gist.prototype.id;
    /** @type {?} */
    Gist.prototype.node_id;
    /** @type {?} */
    Gist.prototype.git_pull_url;
    /** @type {?} */
    Gist.prototype.git_push_url;
    /** @type {?} */
    Gist.prototype.html_url;
    /** @type {?} */
    Gist.prototype.files;
    /** @type {?} */
    Gist.prototype.public;
    /** @type {?} */
    Gist.prototype.created_at;
    /** @type {?} */
    Gist.prototype.updated_at;
    /** @type {?} */
    Gist.prototype.description;
    /** @type {?} */
    Gist.prototype.comments;
    /** @type {?|undefined} */
    Gist.prototype.user;
    /** @type {?} */
    Gist.prototype.comments_url;
    /** @type {?} */
    Gist.prototype.owner;
    /** @type {?} */
    Gist.prototype.forks;
    /** @type {?} */
    Gist.prototype.history;
    /** @type {?} */
    Gist.prototype.truncated;
}
